slack/web/classes/attachments.py

```diff
@@ -0,0 +1,350 @@
+            e in self.MarkdownFields for e in self.markdown_in
+        )
+
+    @JsonValidator(
+        "color attribute must be 'good', 'warning', 'danger', or a hex color code"
+    )
+    def color_valid(self):
+        return (
+            self.color is None
+            or self.color in SeededColors
+            or re.match("^#(?:[0-9A-F]{2}){3}$", self.color, re.IGNORECASE)
+        )
+
+    @JsonValidator("image_url attribute cannot be present if thumb_url is populated")
+    def image_url_and_thumb_url_populated(self):
+        return self.image_url is None or self.thumb_url is None
+
+    @JsonValidator("name must be present if link is present")
+    def author_link_without_author_name(self):
+        return self.author_link is None or self.author_name is not None
+
+    @JsonValidator("icon must be present if link is present")
+    def author_link_without_author_icon(self):
+        return self.author_link is None or self.author_icon is not None
+
+    def get_json(self) -> dict:
+        json = super().get_json()
+        if self.fields is not None:
+            json["fields"] = extract_json(self.fields, AttachmentField)
+        if self.markdown_in:
+            json["mrkdwn_in"] = self.markdown_in
+        return json
+
+
+class BlockAttachment(Attachment):
+    attributes = {"color"}
+    blocks: List[Block]
+
+    def __init__(self, *, blocks: Iterable[Block], color: str = None):
+        """
+        A bridge between legacy attachments and blockkit formatting - pass a list of
+        Block objects directly to this attachment.
+
+        https://api.slack.com/reference/messaging/attachments#fields
+
+        :param blocks: a sequence of Block objects
+
+        :param color: Changes the color of the border on the left side of this
+            attachment from the default gray. Can either be one of good (green), warning
+            (yellow), danger (red), or any hex color code (eg. #439FE0)
+        """
+        super().__init__(text="", color=color)
+        self.blocks = list(blocks)
+
+    def get_json(self) -> dict:
+        json = super().get_json()
+        json.update({"blocks": extract_json(self.blocks, Block)})
+        return json
+
+
+class InteractiveAttachment(Attachment):
+    @property
+    def attributes(self) -> Set[str]:
+        return super().attributes.union({"callback_id"})
+
+    def __init__(
+        self,
+        *,
+        actions: List[Action],
+        callback_id: str,
+        text: str,
+        fallback: str = None,
+        fields: Iterable[AttachmentField] = None,
+        color: str = None,
+        markdown_in: List[str] = None,
+        title: str = None,
+        title_link: str = None,
+        pretext: str = None,
+        author_name: str = None,
+        author_link: str = None,
+        author_icon: str = None,
+        image_url: str = None,
+        thumb_url: str = None,
+        footer: str = None,
+        footer_icon: str = None,
+        ts: int = None,
+    ):
+        """
```

Comment:
The docstring/param list on these two classes is a bit ridiculous, but there's not really any meta-information to keep track of that helps make a builder class worthwhile, IMO

Reply:
On consideration, I _could_ see some utility to an AttachmentBuilder - something like:
```
AttachmentBuilder()
  .text("Some attachment")
  .footer("footer", icon="icon url")
  .author("jim bob", link="jimbob.com", icon="jimbob.com/picture.jpg")
  .field("*Score*", "20")
  .field("*Series*", "Some book series")
  .linkbutton("button", "Click me!", "http://google.com")
  .user_selector()
  .callback_id("abcdefg")
  .build()
```

But, that also ties into the point that pure attachments shouldn't really be encouraged - although I do hope either blocks get a color attribute soon or BlockAttachments never go away :crossed_fingers:

Reply:
I like the `AttachementBuilder`!

Reply:
While I agree an `AttachementBuilder` would be nice, I would not recommend investing too much time into it. Block Kit is the focus and will continue to see new features added.